googlereader/handler.go

```diff
@@ -0,0 +1,163 @@
+package googlereader // import "miniflux.app/googlereader"
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"miniflux.app/http/request"
+	"miniflux.app/http/response/json"
+	"miniflux.app/logger"
+	"miniflux.app/model"
+	"miniflux.app/storage"
+)
+
+type handler struct {
+	store *storage.Storage
+}
+
+const (
+	ReadingList = "user/-/state/com.google/reading-list"
+	Read        = "user/-/state/com.google/read"
+	Starred     = "user/-/state/com.google/starred"
+)
+
+const (
+	StreamIdParam        = "s"
+	StreamExcludeIdParam = "xt"
+	StreamMaxItems       = "n"
+)
+
+// Serve handles Google Reader API calls.
+func Serve(router *mux.Router, store *storage.Storage) {
+	handler := &handler{store}
+	middleware := newMiddleware(store)
+	router.HandleFunc("/accounts/ClientLogin", middleware.clientLogin).Methods(http.MethodPost).Name("ClientLogin")
+	sr := router.PathPrefix("/reader/api/0").Subrouter()
+	sr.Use(middleware.handleCORS)
+	sr.Use(middleware.apiKeyAuth)
+	sr.Methods(http.MethodOptions)
+	sr.HandleFunc("/token", middleware.token).Methods(http.MethodGet).Name("Token")
+
+	sr.HandleFunc("/user-info", handler.userInfo).Methods(http.MethodGet).Name("UserInfo")
+	sr.HandleFunc("/subscription/list", handler.subscriptionList).Methods(http.MethodGet).Name("SubscriptonList")
+	sr.HandleFunc("/stream/items/ids", handler.streamItemIDs).Methods(http.MethodGet).Name("StreamItemIDs")
+	sr.PathPrefix("/").HandlerFunc(handler.serve).Methods(http.MethodPost, http.MethodGet).Name("GoogleReaderApiEndpoint")
+}
+
+func (h *handler) subscriptionList(w http.ResponseWriter, r *http.Request) {
+	userID := request.UserID(r)
+	clientIP := request.ClientIP(r)
+
+	logger.Info("[Reader][subscription/list][ClientIP=%s] Incoming Request for userID  #%d", clientIP, userID)
+
+	var result subscriptionsResponse
+	feeds, err := h.store.Feeds(userID)
+	if err != nil {
+		json.ServerError(w, r, err)
+		return
+	}
+	result.Subscriptions = make([]subscription, 0)
+	for _, feed := range feeds {
+		category, err := h.store.Category(userID, feed.Category.ID)
```

Comment:
This query should not be necessary. The feed category should already be present in the list of feeds.

https://github.com/miniflux/v2/blob/1c9f0005763f73376d767639fd989d389b6f0e53/storage/feed_query_builder.go#L169-L170